### DISCUSSION TITLE
Indent subcirquits in subcirquits?

### DISCUSSION THREAD

**Discussion by Felguja** [logisim-evolution/logisim-evolution#1592] [Ideas]
On more complex projects, the list of subcirquits starts to grow and becomes a little overwhelming to keep up with, would it be a good idea to add an indentation feature to keep subcirquits inside other subcirquits organized?

**Comment by davidhutchens** [comment]
If you switch from Design to Simulate mode (located above the list of subcircuits and built-in components), it will give you a hierarchical view of what circuits are placed inside another circuit. It shows multiple copies when it is placed multiple times. Is this the view you wish to see? This view is intended to allow you to work your way down the circuits to view individual parts during simulation.

Another way to organize parts is to build separate .circ files and include them as libraries in larger things. Would that help you?

If those suggestions are not helpful, could you describe more precisely what you are suggesting?

**Reply by Felguja** [reply]
I am aware of the Simulate view, and that indentation is exactly what i was thinking about, it makes it more intuitive to know where circuits go. What i was thiking was exactly that, but for the Design view, except without the duplicates for multiple instances of the same cirquit

Altough i was not aware of the libraries function, will mess around with it.

These suggestions are incredibly helpfull, thank you
